packages/swingset-liveslots/src/stop-vat.js

```diff
@@ -285,10 +288,15 @@ export async function releaseOldState(tools) {
   // refcount decrements which may drop some virtuals from the DB. It
   // might also drop some objects from RAM.
 
-  const abandonedVrefSet = new Set();
-  identifyExportedRemotables(abandonedVrefSet, tools);
-  identifyExportedFacets(abandonedVrefSet, tools);
-  abandonExports(abandonedVrefSet, tools);
+  // TODO: Decide how much (if any) cleanup to do here in the vat.
+  // The kernel simulates abandonExports as part of vat upgrade,
+  // but does not decrement vat-side refcounts which could allow us to
+  // drop durable objects that were only being kept alive by references from
+  // non-durable objects.
```

Comment:
@warner any specific thoughts on this?

Reply:
My current thinking is that we punt on this until the future. I suspect we won't have a lot of virtual->durable or ephemeral->durable edges in the reference graph, and that userspace will only be using durables for things that really represent important long-term state. But sooner or later our GC story needs to improve, and a full mark-and-sweep scheme is probably in our future, at which point we'll be able to clean up the lingering durables that this fails to catch today.